recipes/libblst/meta.yaml

```diff
@@ -0,0 +1,129 @@
+{% set name = "libblst" %}
+{% set version = "0.3.13" %}
+
+{% set so_major_version = version.split('.')[0] %}
+package:
+  name: libblst-split
+  version: {{ version }}
+
+source:
+  url: https://github.com/supranational/blst/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 89772cef338e93bc0348ae531462752906e8fa34738e38035308a7931dd2948f
+  patches:
+    - patches/0001-unix-add-libversion.patch  # [unix]
+    - patches/0001-win-simplify-build.sh.patch  # [win]
+
+build:
+  number: 0
+  script_env:
+    - PKG_MAJOR_VERSION={{ so_major_version }}
+
+requirements:
+  build:
+    - {{ compiler('c') }}  # [not win]
+    - {{ compiler('m2w64_c') }}  # [win]
+    - {{ stdlib('c') }}  # [not win]
+    - {{ stdlib('m2w64_c') }}  # [win]
+    - libxcrypt  # [unix]
+    - llvm-tools  # [win]
+    - sed  # [unix]
+
+outputs:
+  - name: libblst
+    files:
+      - lib/libblst.so  # [linux]
+      - lib/libblst.dylib  # [osx]
+      - Library\\lib\\blst.lib  # [win]
+
+      - lib/pkgconfig/blst.pc  # [unix]
+      - Library\\lib\\pkgconfig\\blst.pc  # [win]
+      - lib/pkgconfig/libblst.pc  # [unix]
+      - Library\\lib\\pkgconfig\\libblst.pc  # [win]
+    build:
+      run_exports:
+        - {{ pin_subpackage('libblst-' ~ so_major_version) }}
+    requirements:
+      build:
+        - {{ stdlib('c') }}  # [not win]
+        - {{ stdlib('m2w64_c') }}  # [win]
+      host:
+        - {{ pin_subpackage('libblst-' ~ so_major_version, exact=True) }}
+        - {{ pin_subpackage('libblst-headers-' ~ so_major_version, exact=True) }}
+      run:
+        - {{ pin_subpackage('libblst-' ~ so_major_version, exact=True) }}
+        - {{ pin_subpackage('libblst-headers-' ~ so_major_version, exact=True) }}
+    test:
+      commands:
+        - test -f ${PREFIX}/include/blst.h  # [unix]
+        - test -f ${PREFIX}/include/blst_aux.h  # [unix]
+        - if not exist %LIBRARY_INC%\\blst.h exit 1  # [win]
+        - if not exist %LIBRARY_INC%\\blst_aux.h exit 1  # [win]
+
+        - test -f ${PREFIX}/lib/libblst.so  # [linux]
+        - test -f ${PREFIX}/lib/libblst.dylib  # [osx]
+        - if not exist %LIBRARY_LIB%\\blst.lib exit 1  # [win]
+
+
+
+  - name: libblst-headers-{{ so_major_version }}
+    files:
+      - include/blst.h  # [unix]
+      - include/blst_aux.h  # [unix]
+      - Library\\include\\blst.h  # [win]
+      - Library\\include\\blst_aux.h  # [win]
+    test:
+      commands:
+        - test -f ${PREFIX}/include/blst.h  # [unix]
+        - test -f ${PREFIX}/include/blst_aux.h  # [unix]
+        - if not exist %LIBRARY_INC%\\blst.h exit 1  # [win]
+        - if not exist %LIBRARY_INC%\\blst_aux.h exit 1  # [win]
+
+
+
+  - name: libblst-{{ so_major_version }}
```

Comment:
To be sure: They are co-installable?

Reply:
Yes? not sure what you mean/ask, I usually follow that pattern that was taught to me when I make my first c-lib. Also in child packages I try to only install the {{ so }} lib.
Though, do teach  a better way ;P

Reply:
We only use this pattern here if you can install and use `libblst-0` and `libblst-1` at the same time. Otherwise, we would name the package simply `libblst` to avoid accidentally having two incompatible versions in the same process.

Reply:
Oh ... I did not understand that. I really complicate my life for no reason, it was quite hard bending this package (and others to implement the `major_version` scheme!

At this point, do you want me to remove all of it? or would you take pity and let it go? I have 2 packages that rely upon this scheme in the pipeline, meaning depending upon `libblst-0` - Well, I think I know the answer anyway given the potential burden of maintainability on others, I'll get going with the update

The other recipes are `c-kzg-4844` and `blst-py`

Reply:
So, kidding aside, I think it may be useful to maintain the library version. The author mentioned that the lib was not originally intended for "public usage", so it may evolve in ways that may not maintain backward compatibility.

Reply:
@xhochy Could you comment whether this reasoning works for you or if you will only accept this without version number? (I'm interested in it as it is necessary for other package updates)

Reply:
Make it without a version number, otherwise, you would need to do some work on top to ensure that `libblst0` and `libblst1` are not co-installed (which is tricky).

Reply:
It'd be great if you had a bit of time to teach me the principles, or point me to a few recipes. Is the issue that the `0` and `1` may be linked to incompatible external dynamic libraries? I could imagine `ckzg` needing `0` and `blst-py` `1` and having them specifically linked to the appropriate version in their recipe

I'd just like to learn more, if it's just a matter of more work, I did quite a bit already and rather do more than have the many weeks invested be all for naught

In any case, I imagine your time is not that elastic, so I'll get going (with heavy heart) on the refactoring
Thx